### ISSUE TITLE
Feature: package.json "imports" field

### ISSUE THREAD

**Issue by huntie** [facebook/metro#978]
## 👋🏻 Good First Issue

- An issue or task that is ideal for new open source contributors.
- Please reach out to the issue creator before claiming an issue to discuss design.

## Description

[Incoming]

**Comment by siddarthkay** [comment]
Hello @huntie 
I would love to work on adding this feature when you add some more description on the requirement and expectations :)

**Comment by huntie** [comment]
Hey @siddarthkay! I've added a description now. If you're still happy to take this on, let me know :) (and feel free to [DM me on Twitter](https://twitter.com/alxhnt)).

**Comment by siddarthkay** [comment]
@huntie : thanks for the detailed description, for now its good enough to get me started! for more questions I'll paste comments here!
Could you please assign this issue to me?

**Comment by jameslawson** [comment]
@huntie  From the [discussion](https://github.com/facebook/metro/pull/995#issuecomment-1943329549)  in the linked pull request, I think this issue is up for grabs? (cc: @kraenhansen ) Or maybe @kraenhansen was interested in picking it up?

Either way, I've had a go at this myself here: https://github.com/facebook/metro/pull/1302 – please feel free to check it out and see if the changes there are on the right path 😄  - I managed to get some basic tests to pass 🎉

**Comment by robhogan** [comment]
Yes, thanks! And thanks again @jameslawson for the implementation
